public/app/features/alerting/unified/api/buildInfo.ts

```diff
@@ -61,13 +62,22 @@ export async function discoverDataSourceFeatures(dsSettings: {
   // The current implementation of Loki's build info endpoint is useless
   // because it doesn't provide information about Loki's available features (e.g. Ruler API)
   // It's better to skip fetching it for Loki and go the Cortex path (manual discovery)
-  const buildInfoResponse = type === 'loki' ? undefined : await fetchPromBuildInfo(url);
 
+  let fallBackInfoResponse: PromBuildInfoSeriesResponse | undefined;
+  const isLoki = type === 'loki';
+  const buildInfoResponse = isLoki
+    ? undefined
+    : await fetchPromBuildInfo(url).catch(async () => {
+        fallBackInfoResponse = await fetchPromBuildInfoFallback(url);
```

Comment:
Should we move `fetchPromBuildInfoFallback` to be part of the `fetchPromBuildInfo` function? This function is already quite complex and we add more complexity to it
We would also have more control over when to use the fallback and when to skip the error

Reply:
Its tough because they return completely different objects, having `fetchPromBuildInfo` return two different interfaces gunks up the code quite a bit. I'll try to play around with this a bit more and see if there's a better organization